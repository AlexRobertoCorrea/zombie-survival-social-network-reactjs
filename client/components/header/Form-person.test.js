import React from 'react';

import FormPerson from './Form-person';

const getFieldData = () => ({
  label: '',
  id: '',
  rules: '',
  required: false,
  type: ''
});

describe('<FormPerson />', () => {
  const handleSurvivorPerson = () => {};
  const options = {
    form: {},
    handleCloseDialogSurvivor: () => {},
    notifier: {},
    store: {
      peopleStore: {
        fetchingData: false,
        people: [],
        person: {},
        savingData: false,
      }
    }
  };
  const fields = {
    name: getFieldData(),
    age: getFieldData(),
    gender: getFieldData(),
    latitude: getFieldData(),
    longitude: getFieldData(),
    water: getFieldData(),
    food: getFieldData(),
    medication: getFieldData(),
    ammunition: getFieldData()
  };
  const gendersOptions = [{
    label: '',
    value: ''
  }];
  
  const formPerson = shallow(<FormPerson
    handleSurvivorPerson={handleSurvivorPerson}
    options={options}
    fields={fields}
    gendersOptions={gendersOptions}
  />);
  
  it('renders the formPerson as expected', () => {
    expect(formPerson).toMatchSnapshot();
  });
});
