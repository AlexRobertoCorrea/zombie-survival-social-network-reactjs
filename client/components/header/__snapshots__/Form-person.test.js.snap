// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormPerson /> renders the formPerson as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      sm={6}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={Array []}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={3}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={Array []}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={3}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={
          Array [
            Object {
              "label": "",
              "value": "",
            },
          ]
        }
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={6}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={Array []}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={6}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={Array []}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        gutterBottom={true}
        variant="subheading"
      >
        Items
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={3}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={Array []}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={3}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={Array []}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={3}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={Array []}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={3}
      xs={12}
    >
      <InputForm
        field={
          Object {
            "id": "",
            "label": "",
            "required": false,
            "rules": "",
            "type": "",
          }
        }
        form={Object {}}
        maxLength={50}
        selectOptions={Array []}
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</form>
`;
